lib/vagrant/util/which.rb

```diff
@@ -37,11 +37,13 @@ def self.which(cmd, **opts)
           search_path = ENV['PATH']
         end
 
-        SilenceWarnings.silence! do
-          search_path.encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '').split(File::PATH_SEPARATOR).each do |path|
-            exts.each do |ext|
-              exe = "#{path}#{File::SEPARATOR}#{cmd}#{ext}"
-              return exe if File.executable? exe
+        if !search_path.nil?
```

Comment:
I think that instead of including a conditional here, just forcing the type would be a bit cleaner:

```ruby
search_path.to_s.encode('UTF-8', ...
```